keycode.c

```diff
@@ -0,0 +1,293 @@
+/*
```

Comment:
Use SPDX style for license.  See any file in fcitx  as an example

https://github.com/fcitx/fcitx5/blob/e99498a801cc99e5d622844153e59667a1d3acc7/src/frontend/dbusfrontend/dbusfrontend.h#L1